goal_src/engine/camera/cam-layout.gc

```diff
@@ -5,3 +5,5155 @@
+    )
+   (cam-layout-save-campointsoffset
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-tiltAdjust
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-stringMinLength
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-stringMaxLength
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-stringMinHeight
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-stringMaxHeight
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-stringCliffHeight
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-maxAngle
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-campoints-flags
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (cam-layout-save-focalpull-flags
+    arg1
+    (the-as string s4-2)
+    (the-as entity-actor arg0)
+    )
+   (format s4-2 "}~%")
+   (file-stream-close s4-2)
+   )
+  )
+
+(defbehavior clmf-save-one cam-layout ((arg0 symbol))
+  (let ((s5-1 (logtest? (the-as int arg0) 8))
+        (gp-1 (logtest? (the-as int arg0) 16))
+        )
+   (if s5-1
+    (format #t "~%~%~%=================================~%")
+    )
+   (clmf-save-single (-> self cam-entity) s5-1 gp-1)
+   (if s5-1
+    (format #t "===============================~%~%~%~%")
+    )
+   (if gp-1
+    (format
+     #t
+     "'~S' save completed~%"
+     (res-lump-struct
+      (-> self cam-entity)
+      'name
+      structure
+      :time
+      (-> (new 'static 'array float 1 -1000000000.0) 0)
+      )
+     )
+    )
+   )
+  #t
+  )
+
+(defbehavior clmf-save-all cam-layout ((arg0 symbol))
+  (let ((s5-1 (logtest? (the-as int arg0) 8))
+        (gp-1 (logtest? (the-as int arg0) 16))
+        )
+   (if s5-1
+    (format #t "~%~%~%=================================~%")
+    )
+   (let ((v1-5 (-> *camera-engine* alive-list next0)))
+    *camera-engine*
+    (let ((s4-0 (-> v1-5 next0)))
+     (while (!= v1-5 (-> *camera-engine* alive-list-end))
+      (let ((a0-4 (-> (the-as connection v1-5) param1)))
+       (clmf-save-single (the-as entity-camera a0-4) s5-1 gp-1)
+       )
+      (set! v1-5 s4-0)
+      *camera-engine*
+      (set! s4-0 (-> s4-0 next0))
+      )
+     )
+    )
+   (if s5-1
+    (format #t "===============================~%~%~%~%")
+    )
+   (if gp-1
+    (format #t "camera save all completed~%")
+    )
+   )
+  #t
+  )
+
+(deftype clmf-cam-flag-toggle-info (structure)
+  ((key       float  :offset-assert   0)
+   (force-on  int32  :offset-assert   4)
+   (force-off int32  :offset-assert   8)
+   )
+  :method-count-assert 9
+  :size-assert         #xc
+  :flag-assert         #x90000000c
+  )
+
+
+(defbehavior clmf-cam-flag-toggle cam-layout ((arg0 int) (arg1 int))
```

Comment:
there's some messed up bitfield stuff in this function, but probably okay to leave for now